packages/next-on-pages/src/buildApplication/buildMetadataFiles.ts

```diff
@@ -26,22 +26,26 @@ export async function buildMetadataFiles(
  *
  * @param outputDir Output directory for the metadata files.
  */
-async function buildNextStaticHeaders(outputDir: string) {
+async function buildNextStaticHeaders(
+	outputDir: string,
+	opts: BuildMetadataFilesOpts,
+) {
 	const vercelConfig = await getVercelConfig();
 
 	const hitRoutes = getPhaseRoutes(vercelConfig.routes ?? [], 'hit');
 
-	const nextStaticRoute = hitRoutes.find(
-		route => route.src?.startsWith('/_next/static/'),
+	const nextStaticRoute = hitRoutes.find(route =>
+		route.src?.includes('/_next/static/'),
 	);
 	const nextStaticHeaders = (nextStaticRoute as VercelSource)?.headers;
+	const nextStaticPath = getNextStaticDirPath(opts);
```

Comment:
```suggestion
	const nextStaticPath = getNextStaticDirPath(opts);
	const nextStaticRoute = hitRoutes.find(route =>
		route.src?.startsWith(nextStaticPath),
	);
	const nextStaticHeaders = (nextStaticRoute as VercelSource)?.headers;
```

If we're real retrieving the path, it would make sense to use it for the `route.src` check as well.